src/server/rds/benches/benchmark.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! This is a very basic benchmark which tests only get requests with a few
+//! different key and value sizes. It's only using one connection and a very
+//! primitive blocking client, so these results do not reflect the true
+//! performance of the server when under load. It can be used to get a rough
+//! idea of how changes may impact performance.
+//!
+//! For formal performance testing, it is recommended to use
+//! [rpc-perf](https://github.com/twitter/rpc-perf) or another cache
+//! benchmarking tool which supports the Memcache ASCII protocol.
+
+use config::SegcacheConfig;
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+use pelikan_rds::Rds;
+
+use std::io::{Read, Write};
+use std::net::TcpStream;
+use std::time::Duration;
+const RESP_NIL: &[u8; 5] = b"$-1\r\n";
+const RESP_OK: &[u8; 5] = b"+OK\r\n";
+fn get_benchmark(c: &mut Criterion) {
+    // use the default config
+    let config = SegcacheConfig::default();
```

Comment:
It probably shouldn't be called `SegcacheConfig` here... but okay to address in a separate PR.

Reply:
yeah I can possibly fix this in this PR. Let me check what it would require